parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -1274,7 +1276,7 @@ private void readChunkPages(Chunk chunk, BlockMetaData block, ColumnChunkPageRea
     } else { // encrypted column
       rowGroup.addColumn(
           chunk.descriptor.col,
-          chunk.readAllPages(
+          chunk.readAllPages( // @Todo this must be made lazy too?
```

Comment:
I think this needs to be updated too. I'll try to put together a test using Mockito/spies to actually validate the sequence of `readPage()` during individual record materialization

Reply:
I think memory has already been allocated at [readAll](https://github.com/apache/parquet-mr/blob/cf294a35ca4b635f44357fe4c170f3b66870dc6e/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java#L2041C17-L2041C24). 
[This PR](https://github.com/apache/parquet-mr/pull/968/files?diff=split&w=0) may be helpful

Reply:
ahh I hadn't seen that PR, looks really similar. It's still open but hasn't been updated in a year -- is there any plan to merge it?

Reply:
I think for this PR, I'll have to update `ChunkData` so instead of eagerly reading the full stream, it:

1. Initializes by reads enough bytes to verify CRC/load page header.
2. Loads compressed_page_size # of bytes at a time, refilling as needed

Reply:
Thanks Gang, will take a look. This is a pretty complex problem to solve - Steve's PR definitely looks more complete than mine!